pages/api/v1/calendar/event.ts

```diff
@@ -25,10 +37,49 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse<
 
   // Create Google calendar event
   try {
-    await calendar.createCalendarEvent(name, start, end, location, description, eventLink);
-    return res.status(200).json({ message: "Event created successfully!" });
+    const eventID = await calendar.createCalendarEvent(name, start, end, location, description, eventLink);
+    return res.status(200).json({ message: `Event created with calendar ID: ${eventID}` });
+  } catch (err) {
+    return res.status(400).json({ error: JSON.stringify(err) });
+  }
+}
+
+async function patch_handler(req: NextApiRequest, res: NextApiResponse<Data>) {
+  const { eventID, name, start, end, description, location, eventLink } = req.body;
+  
+  // Validate body values exist and are acceptable
+  if (!eventID) return res.status(400).json({ error: "Missing required body field: eventID" });
+  if (!name) return res.status(400).json({ error: "Missing required body field: name" });
+  if (!start) return res.status(400).json({ error: "Missing required body field: start" });
+  if (!end) return res.status(400).json({ error: "Missing required body field: end" });
+  if (!description)
+    return res.status(400).json({ error: "Missing required body field: description" });
+  if (!location) return res.status(400).json({ error: "Missing required body field: location" });
+  if (!eventLink) return res.status(400).json({ error: "Missing required body field: eventLink" });
+  if (new Date(start) < new Date())
+    return res.status(400).json({ error: "Start date cannot be in the past" });
+  if (new Date(end) < new Date())
+    return res.status(400).json({ error: "End date cannot be in the past" });
```

Comment:
Probably makes sense to move this into a helper function `validateCalendarBody` or something if you're using it twice